sqli/sql-injection.go

```diff
@@ -21,50 +22,97 @@ var templates = template.Must(template.ParseFiles(
 	"./views/partials/ruleInfo.gohtml",
 ))
 
-func sqliTemplate(w http.ResponseWriter, r *http.Request, routeInfo utils.Route) (template.HTML, bool) {
+func sqliTemplate(w http.ResponseWriter, r *http.Request, params utils.Parameters) (template.HTML, bool) {
 	var buf bytes.Buffer
 
-	err := templates.ExecuteTemplate(&buf, "sqlInjection", routeInfo)
+	err := templates.ExecuteTemplate(&buf, "sqlInjection", params)
 	if err != nil {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 	}
 	return template.HTML(buf.String()), true
 
 }
 
-func sqlite3Handler(w http.ResponseWriter, r *http.Request, routeInfo utils.Route, splitURL []string) (template.HTML, bool) {
+func headersHandler(w http.ResponseWriter, r *http.Request, routeInfo utils.Route, splitURL []string) (template.HTML, bool) {
+	// Currently we pass only json credentials in headers for SQL Injection
+	if splitURL[3] != "json" {
+		return template.HTML("INVALID URL"), false
+	}
+
+	var credentials struct {
+		Username string `json:"username"`
+		Password string `json:"password"`
+	}
+	err := json.Unmarshal([]byte(r.Header.Get("credentials")), &credentials)
+	if err != nil {
+		log.Printf("Could not parse headers to json, err = %s", err)
+	}
+
+	// setting up a database to execute the built query
+	var sqlite3Database *sql.DB
+	sqlite3Database, err = setupSqlite3()
+	if err != nil {
+		return template.HTML(err.Error()), false
+	}
+	defer func() {
+		_ = sqlite3Database.Close()
+	}()
+
+	query := getSqliteSafetyQuery(sqlite3Database, credentials.Username, splitURL[4])
+
+	_ = os.Remove("tempDatabase.db")
+	return template.HTML(query), false //change to out desired output
```

Comment:
Can you clarify this comment?  "Change to out..."?

Reply:
It seems when sqli-injection was implemented, no specific tables or data was created but just the db file - _tempDatabase.db_. 
I assume that's the reason we just display the query in the response and only track whether it was executed successfully or gives an error to report in the logs.